lesson22/src/js/main.js

```diff
@@ -0,0 +1,241 @@
+                class: `js-sortButtons-Item`,
+                'data-state':button.state
+            }
+        });
+
+        button.state !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    addClickEventToSortTableBody(sortButtonsBox,columnKey,usersData);
+
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassForSortButton = (currentButton,nextButton) => {
+    currentButton.classList.add("hidden");
+    nextButton.classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (category, usersData, state) => {
+    const cloneUsersData = [...usersData];
+
+    switch (state) {
+        case "asc":
+            cloneUsersData.sort((a, b) => a[category] - b[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        case "desc":
+            cloneUsersData.sort((a, b) => b[category] - a[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        default:
+            updateTableBody(usersData);
+    }
+}
+
+const changeState = (currentState) => {
+    switch (currentState) {
+      case "asc":
+        return "desc";
+      case "desc":
+        return "both";
+      default:
+        return "asc";
+    }
+}
+
+const addIsNotClickedForSortButtonsBox = (currentTarget) => {
+    const sortButtons = document.querySelectorAll(".js-sortButtons-Box");
+    for(const button of sortButtons){
+        button.classList.add("is-not-clicked");
+    }
+
+    currentTarget.classList.remove("is-not-clicked");
+}
+
+const resetStatusForSortButton = () => {
+    const sortButtons = document.querySelectorAll(".is-not-clicked > .js-sortButtons-Item");
+    sortButtons.forEach((button) => {
+        const state = button.getAttribute("data-state");
+        (state !== "both") ? button.classList.add("hidden") : button.classList.remove("hidden");
+    });
+}
+
+const addClickEventToSortTableBody = (sortButtonsBox,category,usersData)  => {
+    sortButtonsBox.addEventListener("click", (e) => {
+        if(e.currentTarget === e.target) return;
+
+        addIsNotClickedForSortButtonsBox(e.currentTarget);
+        resetStatusForSortButton();
+        
+        const currentState = e.target.getAttribute("data-state");
+        const nextState  = changeState(currentState);
+        const nextButton = sortButtonsBox.querySelector(`[data-state=${nextState}]`);
+
+        toggleHiddenClassForSortButton(e.target,nextButton);
+        sortUsersData(category, usersData, nextState);
+    });
+} 
```

Comment:
It's so much better now!
If I could only comment on one point.

Usage of usersData.

I think the bug occurs when `usersData` is passed to functions that do not actually use it as an argument, as if it were a bucket relay.

For example, you could define ColumnConfig as tableConfig like this
```js
const tableConfig = {
  column: {
    id: {
      value: "ID",
      hasSort: true
    },
    name: {
      value: "名前",
      hasSort: false
    },
    gender: {
      value: "性別",
      hasSort: false
    },
    age: {
      value: "年齢",
      hasSort: true
    }
  },
  usersData: null  //here!!
};
```
Then we can go to reference against this definition data and there will be no unnecessary relay of arguments.

[codesandbox](https://codesandbox.io/s/lesson22-latest-forked-4lftsz?file=/src/js/main.js)

Reply:
Thanks for the great suggestion.
This way is very smart as it avoids unnecessary relaying of arguments. 
Thanks also for the CodeSandBox. 

I fixed.
https://github.com/nakagawamai/js-lesson/pull/24/commits/633a88a66a4b9167b686c96d7e0c761214504c52